.github/cloud-samples-tools/pkg/utils/runAll.go

```diff
@@ -0,0 +1,55 @@
+/*
+ Copyright 2024 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package utils
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"sync/atomic"
+)
+
+// RunAll runs all the commands in parallel.
+func RunAll(packages []string, script string, maxGoroutines int) int64 {
+	var failures int64
+	guard := make(chan struct{}, maxGoroutines)
+	for _, pkg := range packages {
+		guard <- struct{}{} // block if channel is filled
+		go func() {
+			output, err := exec.Command("bash", script, pkg).CombinedOutput()
+			if err != nil {
+				printError(os.Stderr, pkg, err, string(output))
+				atomic.AddInt64(&failures, 1)
```

Comment:
comment: I think you might want to use `Int64.Add()` here instead, referring to the [docs](https://pkg.go.dev/sync/atomic#AddInt64).

Reading this makes me wonder whether you might want to put a mutex around the critical path here (line 37).

Reply:
Removed this function, so I'm resolving this conversation for now. But this is something we'll have to discuss and address when it gets time to implement this.